.github/workflows/bundle-size.yml

```
@@ -0,0 +1,90 @@
+name: Bundle Size Analysis
+
+on:
+  pull_request:
+    paths:
+      - "packages/core/src/components/**"
+      - "packages/core/package.json"
+      - ".size-limit.js"
+
+permissions:
+  pull-requests: write
+
+jobs:
+  build:
+    name: Build
+    uses: ./.github/workflows/build-and-upload.yml
+    secrets:
+      npm_token: ${{ secrets.npm_token }}
+
+  bundle-size:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - name: Checkout PR
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ci-builds-${{ github.run_id }}-${{ github.run_attempt }}
+          path: packages/
+
+      - name: Run size-limit on PR
+        run: |
+          yarn install --frozen-lockfile
+          node scripts/bundle-check/generate-size-limit-config.js
+          ./node_modules/.bin/size-limit --json > pr.json
```

```suggestion
      - name: Run size-limit on PR
        run: |
          node scripts/bundle-check/generate-size-limit-config.js
          ./node_modules/.bin/size-limit --json > pr.json
```

can't you in `generate-size-limit-config.js` make sure that it won't be inside `node_modules`? that it would make the `pr.json` from the scripts automatically?